### DISCUSSION TITLE
Registry no longer authenticating after #29487

### DISCUSSION THREAD

**Discussion by mikethecalamity** [renovatebot/renovate#30125] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

One of our registries is the container registry on our self-hosted Gitlab. With the changes from #29487, that registry is no longer authenticating properly. My guess is that the `matchHost` is now resolved to `https://registry.gitlab.company.com:443` which does not match the registry of the image which does not have `http://`. I also cannot add `http://` to the `Dockerfile` because then the image does not resolve.

Is this a bug? Or is there something in my config I need to do to fix this?

**Config**
```json
"hostRules": [
    {
        "hostType": "docker",
        "matchHost": "registry.gitlab.company.com:443",
        "username": "bot",
        "password": "***********"
    }
]
```

**Dockerfile**
```docker
FROM registry.gitlab.company.com:443/project/image
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
That matchHost value in your time was and still is an invalid value. It needs to be either a host name, a domain name, or a URL. Probably best to just drop the port part.

**Reply by mikethecalamity** [reply]
I tried with and without `https://` in the `matchHost`, neither work.

**Reply by rarkins** [reply]
Then it's the wrong value
